File: leetcode/Python/bitwise-and-of-numbers-range.py
class Solution:
    def rangeBitwiseAnd(self, left: int, right: int) -> int:
        sol, idx = 0, 1<<30
        while idx > 0:
            if idx > right:
                pass
            elif idx > left:
                break
            else:
                left, right, sol = left - idx, right - idx , sol | idx
            idx>>=1
        return sol

File: leetcode/C++/bitwise-and-of-numbers-range.cpp
class Solution {
public:
    int rangeBitwiseAnd(int left, int right) {
        int sol=0;
        for(int i=(1<<30); i>0; i>>=1){
            if(i>right) continue;
            else if(i>left) break;
            left-=i; right-=i; sol|=i;
        }
        return sol;
    }
};